packages/endomoat/src/default-attenuator.js

```diff
@@ -1,62 +1,56 @@
+import { ENDO_ROOT_POLICY as ROOT_POLICY } from './constants.js'
 import {
-  ENDO_ROOT_POLICY as ROOT_POLICY,
-  ENDO_WRITE_POLICY as WRITE_POLICY,
-} from './constants.js'
+  copyWrappedGlobals,
+  getEndowmentsForConfig,
+  // @ts-expect-error - needs exports
+} from 'lavamoat-core/src/endowmentsToolkit.js'
 
-const { assign, create, keys, fromEntries, entries, defineProperties } = Object
+const { keys, fromEntries, defineProperties, getOwnPropertyDescriptors } =
+  Object
 
-let globalOverrides = create(null)
+/** @type {object} */
+let rootCompartmentGlobalThis
 
 export default {
   /**
    * @type {import('@endo/compartment-mapper').GlobalAttenuatorFn<[import('./policy-converter.js').LavaMoatPackagePolicy['globals']]>}
    */
-  attenuateGlobals(params, originalObject, globalThis) {
+  attenuateGlobals(params, originalGlobalThis, packageCompartmentGlobalThis) {
     const policy = params[0]
     if (!policy) {
       return
     }
     console.debug('attenuateGlobals called', params)
     if (policy === ROOT_POLICY) {
-      assign(globalThis, originalObject)
-      // This assumes that the root compartment is the first to be attenuated
-      globalOverrides = globalThis
-      return
-    }
-    if (policy === WRITE_POLICY) {
-      throw new Error('Not yet implemented')
-    }
-    defineProperties(
-      globalThis,
-      fromEntries(
-        entries(policy)
-          .filter(([, policyValue]) => Boolean(policyValue))
-          .map(([key, policyValue]) => {
-            /** @type {PropertyDescriptor} */
-            const spec = {
-              configurable: false,
-              enumerable: true,
-              get() {
-                console.log('- get', key)
-                return (
-                  globalOverrides[key] ??
-                  originalObject[
-                    /** @type {keyof typeof originalObject} */ (key)
-                  ]
-                )
-              },
-            }
-            if (policyValue === WRITE_POLICY) {
-              spec.set = (value) => {
-                console.log('- set', key)
-                globalOverrides[key] = value
-              }
-            }
-            return [key, spec]
-          })
+      rootCompartmentGlobalThis = packageCompartmentGlobalThis
+      // ^ this is a little dumb, but only a little - assuming importLocation is called only once in parallel. The thing is - even if it isn't, the new one will have a new copy of this module anyway.
```

Comment:
I think I see what you're saying, but:

1. Why not `rootCompartmentGlobalThis =?? packageCompartmentGlobalThis`?
2. Why mention "called _once_ in parallel"? 
3. Why would this branch be executed multiple times _at all_?

Reply:
this branch would only be executed once per importLocation call, but there's nothing stopping you to all it multiple times in one process. It'd become problematic if we called it multiple times in parallel while using a shared instance of the attenuator.

Reply:
Would it be worthwhile to guard against this?  e.g.,

```js
function setRootCompartmentGlobalThis(obj) {
  rootCompartmentGlobalThis ??= obj
}

// or

function setRootCompartmentGlobalThis(obj) {
  if (rootCompartmentGlobalThis && obj !== rootCompartmentGlobalThis) {
    throw new ReferenceError('ya dun goofed')
  }
  rootCompartmentGlobalThis ??= obj
}

// and in attenuateGlobals
if (policy === ROOT_POLICY) {
  setRootCompartmentGlobalThis(packageCompartmentGlobalThis)
}
```